### PR TITLE
Added Requirements section to encoding spec

### PR DESCRIPTION
**Pull request by KrzysztofCwalina**
cc: @ahsonkhan, @joshfree, @terrajobst, @davidfowl, @shiftylogic

### CODE REVIEW

File: docs/specs/encoding.md [hunk 1]
```diff
@@ -6,4 +6,13 @@ The existing .NET encoding APIs (System.Text.Encoding) don't work super well in
 2. They don't work with Span\<T\>
 3. They have the overhead of virtual calls (which might be significan to transcoding small slices)
 
+The requirements for the new APIs are as follows:
+
+1. Transcode bytes contained in ReadOnlySpan<byte> into a passed in Span<byte>
+2. The API needs to handle running out of space in the output buffer and then continuing when an additional output buffer is passed in.
+3. The API needs to be fast (on par with native code implementations)
+4. Needs to do this with zero GC allocations
+5. Needs to be stateless (multithreaded)
```
Comment (davidfowl) [inline docs/specs/encoding.md]: I'd love to understand how when we have requirement 2.

Comment (jkotas) [inline docs/specs/encoding.md]: It would tell you how much it consumed from the input buffer.

Comment (davidfowl) [inline docs/specs/encoding.md]: Which component is stateful? The one consuming this? What does that look like?

Comment (KrzysztofCwalina) [inline docs/specs/encoding.md]: The encoding APIs need to be stateless so we can call them on multiple threads without the need to allocate on every thread.

File: docs/specs/encoding.md [hunk 2]
```diff
@@ -6,4 +6,13 @@ The existing .NET encoding APIs (System.Text.Encoding) don't work super well in
 2. They don't work with Span\<T\>
 3. They have the overhead of virtual calls (which might be significan to transcoding small slices)
```
Comment (joshfree) [inline docs/specs/encoding.md]: typo:  significan

Comment (KrzysztofCwalina) [inline docs/specs/encoding.md]: fixed

**Pull request merged**